### PR TITLE
Add email only if the verification is complete

### PR DESCRIPTION
**Pull request by PunitLodha**
Fixes [element-web#16414](https://github.com/vector-im/element-web/issues/16414)
Signed-off-by: Punit Lodha punitlodha@pm.me

Should the input field be cleared when the dialog box is dismissed, or should it be left unchanged with the entered email?
Currently the input is being cleared, but i think we should leave the email, when the dialog is dismissed by the user itself

### CODE REVIEW

File: src/components/views/settings/account/EmailAddresses.js [hunk 1]
```diff
@@ -178,19 +178,21 @@ export default class EmailAddresses extends React.Component {
...
+                    ...this.props.emails,
+                    { address: email, medium: "email" },
+                ];
+                this.props.onEmailsChange(emails);
+            }
             this.setState({
                 addTask: null,
                 continueDisabled: false,
                 verifying: false,
                 newEmailAddress: "",
```
Comment (jryans) [inline src/components/views/settings/account/EmailAddresses.js:194]: As you mentioned in the PR, seems like a nice refinement to leave the new email in the input if validation was not finished.

**Review by jryans — CHANGES_REQUESTED** [review]
Thanks for looking into this! 😄 

I believe the following components also have similar bugs:

* `settings/account/PhoneNumbers.js`
* `settings/discovery/EmailAddresses.js`
* `settings/discovery/PhoneNumbers.js`

Could you fix up these as well?

File: src/components/views/settings/account/EmailAddresses.js (hunk 1 above)
Comment (PunitLodha) [reply src/components/views/settings/account/EmailAddresses.js:194]: Yes, I'll change that

Comment (PunitLodha) [reply src/components/views/settings/account/EmailAddresses.js]: Done

**Comment by PunitLodha** [comment]
`settings/account/PhoneNumbers.js` is fixed, but i couldn't find a way to test the discovery components. Do i have to accept the Discovery conditions for these components to show up?

**Comment by jryans** [comment]
> `settings/account/PhoneNumbers.js` is fixed, but i couldn't find a way to test the discovery components. Do i have to accept the Discovery conditions for these components to show up?

Yes, unless you modify things for testing... Here's the code involved:

https://github.com/matrix-org/matrix-react-sdk/blob/5c1b38a48cae9bf3511b36d2b0a417b45a953fc4/src/components/views/settings/tabs/user/GeneralUserSettingsTab.js#L309-L330

You could run your own local identity server and configure a local Element to use to that if you don't want to share any data with someone else's server.

If you'd rather not do any of that, we could leave the discovery parts out of this PR. Please click the re-request review wheel at the top right whenever you're ready.

**Comment by PunitLodha** [comment]
I checked the discovery components, and this issue doesn't exist there as there is no password verification for discovery components

**Review by jryans — APPROVED** [review]
Thanks for working on this, looks good to me! 😄

**Pull request merged**